### ISSUE TITLE
Software Architecture 

### ISSUE THREAD

**Issue by UmangBR** [amosproj/amos2022ss03-turtlebot-fleet-management#4]
## User story
1. As a team
2. We need a Software Architechture description by the end of sprint 2
3. So that everyone has clarity on all the software components 

## Acceptance criteria
*Write an overview diagram of runtime components (one page)
*Draw an overview diagram of code components (one page)
*Write the summary of the underlying technology stack (one page)
*Write a textual explanation of the diagrams and choices (one page)

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
